src/app/services/wallet.service.ts

```diff
@@ -57,19 +59,22 @@ export class WalletService {
 
   sendSkycoin(wallet: Wallet, address: string, amount: number) {
     const addresses = wallet.addresses.map(a => a.address).join(',');
+
     return this.apiService.getOutputs(addresses).flatMap((outputs: Output[]) => {
-      const totalCoins = parseInt((outputs.reduce((count, output) => count + output.coins, 0) * 1000000) + '', 10);
-      const totalHours = outputs.reduce((count, output) => count + output.hours, 0);
+      const minRequiredOutputs =  this.getMinRequiredOutputs(amount, outputs);
+      const totalCoins = parseInt((minRequiredOutputs.reduce((count, output) =>
+        count + output.coins, 0) * 1000000) + '', 10);
+      const totalHours = minRequiredOutputs.reduce((count, output) => count + output.hours, 0);
       const changeCoins = totalCoins - amount * 1000000;
-      const changeHours = totalHours / 4;
+      const changeHours = Math.floor(totalHours * this.allocationRatio);
       const txOutputs: TransactionOutput[] = [{ address: address, coins: amount * 1000000, hours: changeHours }];
```

Comment:
Please use `parseInt` with the coins. Same reason as with line 68.

Also, if `changeCoins == 0` then `hours` should be `changeHours * 2`, filtered with `parseInt`. This is because, if it is not possible to return the remaining hours to the user, they should be sent to the recipient.

Reply:
In case we have a wallet with coins: 1, hours: 10 and want to send 1 coin.
We need to have one output with coins: 1 and hours: ...
a) 10/4 = 2.5 => 2, and since there is no change => 2*2 = 4
b) 10/2 = 5 => 5
Could you, please, clarify what is the correct answer (4 or 5)?

Reply:
The correct answer is 5. In fact, if the example had an output with 2 coins, instead of 1, and the user wanted to send just 1 coin, the most appropriate thing would be to send 2 hours to the recipient and return 3 to the user.